servicemanager/servicemanager.go

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright (C) 2022 Renesas Electronics Corporation.
+// Copyright (C) 2022 EPAM Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package servicemanager
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/aoscloud/aos_common/aoserrors"
+	"github.com/aoscloud/aos_common/image"
+	"github.com/aoscloud/aos_common/utils/action"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/aoscloud/aos_servicemanager/config"
+	"github.com/aoscloud/aos_servicemanager/utils/imageutils"
+	"github.com/aoscloud/aos_servicemanager/utils/uidgidpool"
+)
+
+/**********************************************************************************************************************
+ * Consts
+ *********************************************************************************************************************/
+
+const maxConcurrentActions = 10
+
+/**********************************************************************************************************************
+ * Types
+ *********************************************************************************************************************/
+
+// ServiceStorage provides API to create, remove or access services DB.
+type ServiceStorage interface {
+	GetService(serviceID string) (ServiceInfo, error)
+	GetAllServices() ([]ServiceInfo, error)
+	AddService(ServiceInfo) error
+}
+
+// ServiceManager instance.
+type ServiceManager struct {
+	servicesDir         string
+	downloadDir         string
+	actionHandler       *action.Handler
+	gidPool             *uidgidpool.IdentifierPool
+	serviceInfoProvider ServiceStorage
+}
+
+// ServiceInfo service information.
+type ServiceInfo struct {
+	ServiceID       string
+	AosVersion      uint64
+	ServiceProvider string
+	Description     string
+	ImagePath       string
+	GID             int
+	ManifestDigest  []byte
+	IsActive        bool
+}
+
+/**********************************************************************************************************************
+ * Vars
+ *********************************************************************************************************************/
+
+// ErrNotExist not exist service error.
```

Comment:
Add comment for each error. Or one comment for all.

Reply:
added